src/linker/utilities/general_utils.py

```diff
@@ -90,23 +88,18 @@ def _add_logging_sink(
         )
 
 
-def prepare_results_directory(
-    output_dir: Optional[str], timestamp: bool, config: Config
-) -> Path:
-    results_dir = _generate_results_dir_name(output_dir, timestamp)
+def create_results_directory(output_dir: Optional[str], timestamp: bool) -> Path:
+    launch_time = datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
```

Comment:
nit: put the definition of `launch_time` inside the if block, as it's not used otherwise.